src/components/activities/pharmacyActivity/pharmaceutical/components/pharmaceuticalExpiringDialog/PharmaceuticalExpiringDialog.tsx

```diff
@@ -0,0 +1,191 @@
+import { Autocomplete, TextField } from "@mui/material";
+import Dialog from "@mui/material/Dialog";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import Button from "components/accessories/button/Button";
+import React, { FunctionComponent, useState } from "react";
+import { useTranslation } from "react-i18next";
+import "./styles.scss";
+
+const PharmaceuticalExpiringDialog: FunctionComponent<{
+  isOpen: boolean;
+  handleSecondaryButtonClick: () => void;
+  handlePrimaryButtonClick: (
+    period: string | null,
+    month: string | null
+  ) => void;
+}> = ({ isOpen, handlePrimaryButtonClick, handleSecondaryButtonClick }) => {
+  const { t } = useTranslation();
+
+  const [monthSelected, setMonthSelected] = useState<string | null>(null);
+
+  const [periodSelected, setPeriodSelected] = useState<string | null>(null);
+
+  const periodOptions = [
+    {
+      label: t("pharmacy.stock.expiring.options.today"),
+      value: "TODAY",
+    },
+    {
+      label: t("pharmacy.stock.expiring.options.thisMonth"),
+      value: "THISMONTH",
+    },
+    {
+      label: t("pharmacy.stock.expiring.options.nextMonth"),
+      value: "NEXTMONTH",
+    },
+    {
+      label: t("pharmacy.stock.expiring.options.nextTwoMonths"),
+      value: "NEXTTWOMONTHs",
+    },
+    {
+      label: t("pharmacy.stock.expiring.options.specificMonth"),
+      value: "SPECIFICMONTH",
+    },
+  ];
```

Comment:
Memorize this and pass t as dependency
Create enum and avoid hard coded stuff like above

```suggestion
  const periodOptions = useMemo(() => [ExperingPeriod.TODAY, ...].map(period => ({label: t(`pharmacy.stock.expiring.options.${period.toLowerCase()}`), value: period})), [t]);
```